pkg/actuators/machine/actuator_test.go

```diff
@@ -1263,3 +1259,221 @@ func awsClientBuilderFunc(c awsclient.Client) awsclient.AwsClientBuilderFuncType
 		return c, nil
 	}
 }
+
+func TestPatchMachine2(t *testing.T) {
+	// BEGIN: Set up test environment
+	g := NewWithT(t)
+
+	testEnv := &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join("..", "..", "..", "config", "crds")},
+	}
+
+	var err error
+	cfg, err := testEnv.Start()
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cfg).ToNot(BeNil())
+	defer func() {
+		g.Expect(testEnv.Stop()).To(Succeed())
+	}()
+
+	mgr, err := manager.New(cfg, manager.Options{
+		Scheme:             scheme.Scheme,
+		MetricsBindAddress: "0",
+	})
+	g.Expect(err).ToNot(HaveOccurred())
+
+	actuator, err := initActuator(mgr, t)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	doneMgr := make(chan struct{})
+	go func() {
+		g.Expect(mgr.Start(doneMgr)).To(Succeed())
+	}()
+	defer close(doneMgr)
+
+	// END: setup test environment
+
+	k8sClient := mgr.GetClient()
+
+	awsCredentialsSecret := stubAwsCredentialsSecret()
+	g.Expect(k8sClient.Create(context.TODO(), awsCredentialsSecret)).To(Succeed())
+
+	userDataSecret := stubUserDataSecret()
+	g.Expect(k8sClient.Create(context.TODO(), userDataSecret)).To(Succeed())
+
+	codec, err := providerconfigv1.NewCodec()
+	if err != nil {
+		t.Fatalf("unable to build codec: %v", err)
+	}
+
+	testCases := []struct {
+		name   string
+		mutate func(*machinev1.Machine)
+		expect func(*machinev1.Machine) error
+	}{
+		{
+			name: "Test changing labels",
+			mutate: func(m *machinev1.Machine) {
+				m.ObjectMeta.Labels["testlabel"] = "test"
+			},
+			expect: func(m *machinev1.Machine) error {
+				if m.ObjectMeta.Labels["testlabel"] != "test" {
+					return fmt.Errorf("label \"testlabel\" %q not equal expected \"test\"", m.ObjectMeta.Labels["test"])
+				}
+				return nil
+			},
+		},
+		{
+			name: "Test setting phase",
+			mutate: func(m *machinev1.Machine) {
+				failed := "Failed"
+				m.Status.Phase = &failed
+			},
+			expect: func(m *machinev1.Machine) error {
+				if m.Status.Phase == nil {
+					return fmt.Errorf("phase can't be nil")
+				}
+
+				if *m.Status.Phase != "Failed" {
+					return fmt.Errorf("phase %q not equal expected \"Failed\"", *m.Status.Phase)
+				}
+				return nil
+			},
+		},
+		{
+			name: "Test setting provider status",
+			mutate: func(m *machinev1.Machine) {
+				instanceID := "123"
+				instanceState := "running"
+
+				providerStatus := &providerconfigv1.AWSMachineProviderStatus{
+					InstanceID:    &instanceID,
+					InstanceState: &instanceState,
+				}
+
+				status, err := codec.EncodeProviderStatus(providerStatus)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				m.Status.ProviderStatus = status
+			},
+			expect: func(m *machinev1.Machine) error {
+				providerStatus, err := getMachineStatus(codec, m)
+				if err != nil {
+					return fmt.Errorf("unable to get provider status: %v", err)
+				}
+
+				if providerStatus.InstanceID == nil || providerStatus.InstanceState == nil {
+					return fmt.Errorf("instance ID and state can't be nil")
+				}
```

Comment:
I'd be tempted to combine these with the checks below, so that you get the error out saying what was expected in this case, WDYT?

```
if providerStatus.InstanceID == nil || *providerStatus.InstanceID != "123" {
    return fmt.Errorf("instanceID %q not equal expected \"123\"", providerStatus.InstanceID)
}

if providerStatus.InstanceState == nil || *providerStatus.InstanceState != "running" {
    return fmt.Errorf("instanceState %q not equal expected \"running\"", providerStatus.InstanceState)
}
```